File: nysol_python/mcmd/methods/convTex/db/smp/msetstr.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='msetstr'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,date
A,20081202
A,20081204
B,20081203
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
日付計算で必要となる基準日を(2007年01月01日と定義した場合)すべての行に「 ``20070101`` 」という文字列を追加し「基準日」という項目名で出力する。
'''
script['sh_code']='''
msetstr v=20070101 a=基準日 i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.msetstr(v="20070101", a="基準日", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='複数項目を追加'
script['comment']='''
'''
script['sh_code']='''
msetstr v=20070101,20070201 a=基準日1,基準日2 i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.msetstr(v="20070101,20070201", a="基準日1,基準日2", i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='null値項目追加'
script['comment']='''
'''
script['sh_code']='''
msetstr v= a=追加項目 i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.msetstr(v="", a="追加項目", i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/msetstr.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
customer,date
A,20081202
A,20081204
B,20081203
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Calculate the date by setting a reference date  (defined as January 01, 2007)  and add the string “\verb|20070101|” in all lines and save the output as a new column named “ReferenceDate”.
EOF
scp=<<'EOF'
more dat1.csv
msetstr v=20070101 a=ReferenceDate i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Add multiple fields"
comment=<<'EOF'
EOF
scp=<<'EOF'
msetstr v=20070101,20070201 a=RefDate1,RefDate2 i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Add column with null values"
comment=<<'EOF'
EOF
scp=<<'EOF'
msetstr v= a=NewColumn i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)
